open_rarity/scoring/token_feature_extractor.py

```diff
@@ -0,0 +1,44 @@
+from open_rarity.models import Token
+from open_rarity.models.collection import Collection
+from open_rarity.models.token_ranking_features import TokenRankingFeatures
+
+
+class TokenFeatureExtractor:
+    """
+    Utility class that extract features from tokens
+    """
+
+    @staticmethod
+    def extract_unique_attribute_count(
+        token: Token, collection: Collection
+    ) -> TokenRankingFeatures:
+        """This method extracts unique attributes count from the token
+
+        Parameters
+        ----------
+        token : Token
+            The token to extract features from
+        collection : Collection
+            The collection with the attributes frequency counts to base the
+            token trait probabilities on to calculate score.
+
+        Returns
+        -------
+        TokenFeatures
+            Token features wrapper class that contains extracted features
+
+        """
+
+        unique_attributes_count: int = 0
+
+        for string_attribute in token.metadata.string_attributes.values():
+            count = collection.attributes_frequency_counts[
+                string_attribute.name
+            ][string_attribute.value]
```

Comment:
instead of doing this, you can just do
`count = collection.total_tokens_with_attribute(string_attribute)` (it does exactly what you're doing)

Reply:
Decided to go with original change since tests were failing with suggested approach.